v2/jsonpatch.go

```diff
@@ -149,9 +155,6 @@ func makePath(path string, newPart interface{}) string {
 	if path == "" {
 		return "/" + key
 	}
-	if strings.HasSuffix(path, "/") {
```

Comment:
Why is this part not needed?

Reply:
The test case for this is:

```
	emptyKeyA = `{"":[0]}`
	emptyKeyB = `{"":[]}`
```

The patch should be

```
[
    {
        "op": "remove",
        "path": "//0"
    }
]
```

Before this change, we would have the path `/0`. which was not correct.

I am not sure why it was originally added. Removal of this code did not impact any existing test cases, and the fuzz tester should verify we round-trip successfully. So I think the only case this code was hit was for the empty key edge case